### ISSUE TITLE
error: RPC failed; curl 56 Recv failure: Connection was reset from Windows

### ISSUE THREAD

**Issue by SonalSingh-nxg** [git-for-windows/git#4519]
When I try to push my code to remote using windows, I get the following error:
```
error: RPC failed; curl 56 Recv failure: Connection was reset
send-pack: unexpected disconnect while reading sideband packet
fatal: the remote end hung up unexpectedly
Everything up-to-date
```
However, from linux shell, I am able to push the exact same commit to the same repo.

 - [ x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```
$ git --version --build-options

git version 2.39.2.windows.1
cpu: x86_64
built from commit: a82fa99b36ddfd643e61ed45e52abe314687df67
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.22621.1992]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
> type "$env:USERPROFILE\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

Editor Option: VIM
Custom Editor Path:
Default Branch Option:
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Git Pull Behavior Option: Merge
Use Credential Manager: Enabled
Performance Tweaks FSCache: Enabled
Enable Symlinks: Disabled
Enable Pseudo Console Support: Disabled
Enable FSMonitor: Disabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

** insert your response here **

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

CMD and PowerShell

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
git push origin master
```
 - What did you expect to occur after running these commands?

Push should occur successfully to the remote.

 - What actually happened instead?

error: RPC failed; curl 56 Recv failure: Connection was reset
send-pack: unexpected disconnect while reading sideband packet
fatal: the remote end hung up unexpectedly
Everything up-to-date

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

NO, it is happening with all repos

**Comment by rimrul** [comment]
This seems to be related to #4469

**Comment by AjnaVidya-Developer** [comment]
We are not using HTTP/2. We reverted to using HTTP/1.1, still we are getting this error.

**Comment by ldennington** [comment]
First, it's a good idea to verify the stability of your network connection. If there are no problems there, then try the following:

1. Increase the value of [`http.postBuffer`](https://git-scm.com/docs/git-config#Documentation/git-config.txt-httppostBuffer).
2. `git config --global core.compression 0` (see `core.compression` docs [here](https://git-scm.com/docs/git-config#Documentation/git-config.txt-corecompression)).
3. Set the following before doing another `push`:
```
set GIT_TRACE_PACKET=1
set GIT_TRACE=1
set GIT_CURL_VERBOSE=1
```
That will produce some logs that will give you more information about what's going on with the connection.

I'd also recommend trying the new release that contains the fix for #4469 once it's available just in case.

**Comment by dscho** [comment]
> git version 2.39.2.windows.1

Please first try the latest and greatest Git for Windows version.

**Comment by dscho** [comment]
> I'd also recommend trying the new release that contains the fix for #4469 once it's available just in case.

Excellent call-out. Please try v2.42.0-rc0 once it is out (I hope to be able to publish it later today).

**Comment by MatthewSteeples** [comment]
I was having same problem as original author, although strangely enough only when posting to `master`. Other branches worked fine (although it could have been size of commits I suppose, was testing with 72kb on master but 26kb on a branch).

Updating to 2.42.0-rc0 fixed the problem for me

**Comment by dscho** [comment]
> Updating to 2.42.0-rc0 fixed the problem for me

Excellent. Will close this as done.

**Comment by hoegge** [comment]
Does anybody know when 2.42 will be released? Most of my repos cannot be pushed from Windows and I have to do it from WSL all the time. It must affect a lot of people?

**Comment by dscho** [comment]
> Does anybody know when 2.42 will be released?

Why, yes: Git v2.42.0 is expected this coming Monday, August 21st, according to https://tinyurl.com/gitcal, and Git for Windows traditionally follows suit the next day or so.

> Most of my repos cannot be pushed from Windows and I have to do it from WSL all the time.

Then I must wonder why you don't use [the -rc2 version](https://github.com/git-for-windows/git/releases/tag/v2.42.0-rc2.windows.1), if only to make sure that the problems are truly fixed and won't still occur with v2.42.0? After all, I'm not putting in all that effort so that nobody uses those -rc versions.

**Comment by hoegge** [comment]
Thanks @dscho - was not aware of (should have looked better) that there was an RC. Will def. do that in the future.
I will try as soon as I can reproduce the problem again (last was resolved by pulling somebody elses work).
And **great** with release planned for Monday.
